sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala

```diff
@@ -1172,14 +1172,13 @@ case class Unhex(child: Expression, failOnError: Boolean = false)
     nullSafeCodeGen(ctx, ev, c => {
       val hex = Hex.getClass.getName.stripSuffix("$")
       val maybeFailOnErrorCode = if (failOnError) {
-        val format = UTF8String.fromString("BASE64");
         val binaryType = ctx.addReferenceObj("to", BinaryType, BinaryType.getClass.getName)
         s"""
            |if (${ev.value} == null) {
            |  throw QueryExecutionErrors.invalidInputInConversionError(
            |    $binaryType,
            |    $c,
-           |    $format,
+           |    UTF8String.fromString("HEX"),
```

Comment:
uhh, the format value is not correct too. Could you add a test for the error, please. Like:
https://github.com/apache/spark/blob/45e5f2e375bec915e1683e6d2a222488ba831c91/sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala#L60-L71

Reply:
Not sure what you mean. The format here was wrong in addition to the bad codegen, just fixed both. Assume this was a copy paste error from unbase64

Reply:
> Not sure what you mean. The format here was wrong in addition to the bad codegen, just fixed both.

I mean add a test which fails without your second fix: BASE64 -> HEX

Reply:
This was being like constant folded so it was just being eval'd. Update the existing error test to do both codegen and interpreted, and made one for hex as well